File: entities/src/main/java/com/jbrunton/entities/MoviesRepository.java
package com.jbrunton.entities;

import java.util.List;

import io.reactivex.Observable;

public interface MoviesRepository {
    Observable<Movie> getMovie(String movieId);
    Observable<List<Movie>> searchMovies(String query);
    Observable<List<Movie>> nowPlaying();
    Observable<List<Movie>> discoverByGenre(String genreId);
}

File: entities/src/main/java/com/jbrunton/entities/MoviesRepository.kt
package com.jbrunton.entities

import io.reactivex.Observable

interface MoviesRepository {
    suspend fun getMovie(movieId: String): Movie
    fun getMovieRx(movieId: String): Observable<Movie>
    fun searchMovies(query: String): Observable<List<Movie>>
    fun nowPlaying(): Observable<List<Movie>>
    fun discoverByGenre(genreId: String): Observable<List<Movie>>
}
